### DISCUSSION TITLE
Personal Access Token Read Only

### DISCUSSION THREAD

**Discussion by HariSekhon** [community/community#18666] [Other Feature Feedback, Questions, & Ideas]
Is it possible to create a Personal Access Token with Read Only permissions to all APIs, especially private repos?

## Use Case

GitHub Actions environment secret for Terraform Plan in Pull Requests.

If I can generate a Read Only token then I can remove the approval requirement on the environment secret to allow Terraform Plan workflows to automatically run in Pull Requests.

Right now tokens get Full Control of private repositories, which is too risky to allow devs to use without approval, because they could create new workflows in their PRs that would auto-execute with essentially admin permissions over the repos and github org.

Approvals themselves are causing issues as they often get backed up and cause apply workflows in the main branch to stall waiting for them to be approved.

![Screen Shot 2022-06-15 at 16 42 28](https://user-images.githubusercontent.com/2211051/173870067-b17495e6-c64d-4a3d-a6ee-e404fb4fcacb.png)

**Comment by jsoref** [comment]
Depending on what you're doing, you could have a read-only deploy key.

But, we chose to use https://github.com/runatlantis/ to handle terraform. It is possible to configure runatlantis to run terraform in a readonly mode.

**Reply by HariSekhon** [reply]
Yeah Atlantis was my next step... deploy keys are SSH keys, so not suitable for Terraform plans using GitHub API to manage our GitHub Enterprise Organization and repos.

**Comment by HariSekhon** [comment]
This is a commonly requested feature:

#5317

#33057

Looks like it's on the roadmap but has been open for over a year:

https://github.com/github/roadmap/issues/184
